### PR TITLE
SONARPY-2316 Resolve field members of a ClassType

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**
[SONARPY-2316](https://sonarsource.atlassian.net/browse/SONARPY-2316)

[SONARPY-2316]: https://sonarsource.atlassian.net/browse/SONARPY-2316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java [hunk 1]
```diff
@@ -451,10 +453,32 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
...
-      addStaticFieldToClass(lhsName);
+      addStaticFieldToClass(lhsName, PythonType.UNKNOWN);
     });
   }
 
+  @Override
+  public void visitAnnotatedAssignment(AnnotatedAssignment assignmentStatement) {
+    scan(assignmentStatement.assignedValue());
+    scan(assignmentStatement.annotation());
+    scan(assignmentStatement.variable());
```
Comment (thomas-serre-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java:464]: any reason you don't call `super`? Tests in `TypeInferenceV2Test` are still green if I use the call to `super`
```suggestion
    super.visitAnnotatedAssignment(assignmentStatement);
```

File: python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java [hunk 2]
```diff
@@ -451,10 +453,32 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
...
+
+    Optional.ofNullable(assignmentStatement.variable())
+      .filter(NameImpl.class::isInstance)
+      .map(NameImpl.class::cast)
+      .ifPresent(lhsName -> {
+        if (currentType() instanceof ClassType) {
```
Comment (thomas-serre-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java:470]: _question_ this line is to check that you are in a class definition right?

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java [hunk 3]
```diff
@@ -500,24 +500,87 @@ class D(A, C): pass
...
+      assertThat(classType.members()).hasSize(1);
+      var member = classType.members().iterator().next();
+      assertThat(member.name()).isEqualTo("test");
+      assertThat(member.type()).isEqualTo(PythonType.UNKNOWN);
+    });
 
     Expression expr2 = lastExpression("""
       class A:
         test = "hi"
         test = True
```
Comment (thomas-serre-sonarsource) [inline python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java:563]: I suggest that you update the test with type hint to ensure that the type of `test` Symbol is `bool` in the end.
```suggestion
      class A:
        test : str = "hi"
        test : bool = True
```

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java [hunk 4]
```diff
@@ -562,10 +631,13 @@ class B:
         test = "hi"
       class C(A, B):
         pass
-      C.test
```
Comment (thomas-serre-sonarsource) [inline python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java:565]: why do you have to remove the access to the field `test` everywhere?

**Review by thomas-serre-sonarsource — APPROVED** [review]
LGTM! 
I left clarification questions and a few comments. Well done!

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java (hunk 3 above)
Comment (maksim-grebeniuk-sonarsource) [reply python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java:563]: test added

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java (hunk 4 above)
Comment (maksim-grebeniuk-sonarsource) [reply python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java:565]: discussed

File: python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java (hunk 2 above)
Comment (maksim-grebeniuk-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java:470]: discussed

**Review by thomas-serre-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**